oximeter/db/src/model/fields.rs

```diff
@@ -0,0 +1,188 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Methods for reading / writing oximeter fields to the database.
+
+// Copyright 2024 Oxide Computer Company
+
+use crate::native::block::Block;
+use crate::native::block::Column;
+use crate::native::block::DataType;
+use crate::native::block::ValueArray;
+use crate::query::field_table_name;
+use indexmap::IndexMap;
+use oximeter::FieldType;
+use oximeter::FieldValue;
+use oximeter::Sample;
+use std::collections::BTreeMap;
+
+/// Extract `Block`s for all fields in a `Sample`.
+///
+/// This returns a data block for each field table, which can be inserted into
+/// the database.
+pub(crate) fn extract_fields_as_block(
+    sample: &Sample,
+) -> BTreeMap<String, Block> {
+    let mut out = BTreeMap::new();
+    let timeseries_key = crate::timeseries_key(sample);
+    for field in sample.fields() {
+        let field_type = field.value.field_type();
+        let table_name = field_table_name(field_type);
+        let entry = out.entry(table_name).or_insert_with(|| Block {
+            name: String::new(),
+            info: Default::default(),
+            columns: empty_columns_for_field(field_type),
+        });
+
+        // Push the timeseries name, key, and field name.
+        let Ok(ValueArray::String(timeseries_names)) =
```

Comment:
"timeseries_name", "field_name", etc.. are used in a bunch of places. It may be worth making those constants to get compile time guarantees of correctness.

Reply:
Good idea, will do.

Reply:
Done in 27b34687f